### ISSUE TITLE
Early Stopping that triggers before min_epochs is reached increasingly slows training down

### ISSUE THREAD

**Issue by phasesing** [Lightning-AI/pytorch-lightning#18063]
### Bug description

When early stopping triggers before meeting min_epochs it alerts that training will continue, but training speed immediately falls (on my system it immediately falls by nearly half from 250-300 it/s to 150 it/s, and continues to fall in a nearly linear fashion for the five additional epochs to reach min_epochs, ending at 2.7 it/s.

One likely contributor to this slowdown is that lightning spontaneously logs every training step instead of every 50th (default).

I'm mostly using the example lightning code, modified to suit my particular I/O needs (custom dataloader, network params, custom loss function). You should be able to reproduce this by setting a high min_epochs compared to convergence.

### What version are you seeing the problem on?

v2.0

### How to reproduce the bug

```python
'''python
def main():
    loaderclass = loader.PSILok('512k_jammer_run')
    train,test = loaderclass.generate_dataloader(synth_count=512000,batch_size=64)
    model = LitJamFinder()

    trainer = lp.Trainer(min_epochs=50,max_epochs=500,
                         callbacks=[EarlyStopping(monitor='val_loss',patience=10,mode='min')])
    trainer.fit(model,train,test)
'''
```

### Error messages and logs

No errors, just unexpected runtime behavior.

### Environment

Cannot use wget - environment is Win10 Anaconda. Have you got a pip I could use instead?

cc @carmocca @awaelchli

**Comment by thesofakillers** [comment]
Also getting this error, see these WandB logs:

<img width="655" alt="image" src="https://github.com/Lightning-AI/lightning/assets/26286291/a564fe66-4f03-4300-a732-ec976a2169d5">

**Comment by awaelchli** [comment]
Hello @phasesing @thesofakillers 
Thanks for the report.
Would it be possible for you to provide a runnable code example? If I have that, I can definitely look into it. See for example this comment here https://github.com/Lightning-AI/lightning/issues/15708#issuecomment-1425909297 where I left a code example that easily reproduced the bug that was later fixed.

**Comment by thesofakillers** [comment]
I should note that my case was for 1.8, upgrading to 1.9.2 fixed it.

**Comment by awaelchli** [comment]
This was fixed by #16719. The reason for the slowdown was an excessive triggering of additional validation runs.
